pliers/extractors/text_encoding.py

```diff
@@ -0,0 +1,812 @@
+            
+            embedding_vectors.append(embedding_vector)
+            
+        num_dims = embedding_vectors[0].shape[0]
+        features = ['%s%d' % (self.prefix, i) for i in range(num_dims)]
+        
+        return ExtractorResult(embedding_vectors,
+                               stim,
+                               self,
+                           features=features)
+
+
+class DANExtractor(DirectSentenceExtractor):
+    
+    '''Currently we have DAN (Deep Averaging Networks)
+        encoding as a service running on Tensorflow Hub.
+    '''
+    _hub_path = "https://tfhub.dev/google/universal-sentence-encoder/2"
+    _method = 'dan'
+
+    
+    def __init__(self):
+        
+        super(DANExtractor, self).__init__()
+        self._hub_path = os.path.join(self._embedding_model_path, 'tfhub')
+        os.environ["TFHUB_CACHE_DIR"] = self._hub_path
+        self.dan_encoder = hub.Module(self._hub_path)
+    
+    def setOSEnv(self,path):
+        '''to use TF hub we need to se the cache'''
+        os.environ["TFHUB_CACHE_DIR"] = path
+    
+    def getVector(self,stim,cbow=False):
+        
+        if not isinstance(stim,list):
+            stim = [stim]
+        
+        if cbow == True:
+            stim_cbows = self.generateCbows(stim)
+            stim = stim_cbows
+       
+        embeddings = self.dan_encoder(stim,as_dict=True)
+        with tf.Session() as sess:
+            sess.run(tf.global_variables_initializer())
+            sess.run(tf.tables_initializer())
+            embedding_vectors = sess.run(embeddings)    
+        
+        num_dims = (embedding_vectors.get('default')[0]).shape[0]
+        features = ['%s%d' % (self.prefix, i) for i in range(num_dims)]
+        
+        embedding_dan_vectors = embedding_vectors.get('default')
+        return ExtractorResult(embedding_dan_vectors,
+                               stim,
+                               self,
+                               features=features)
+        
+class BertExtractor(DirectSentenceExtractor):
+      
+    '''For more information regarding Bert please
+          refer to the original paper - 
+          BERT: Pre-training of Deep Bidirectional Transformers for
+        Language Understanding (Devlin et al.) and
+        code  - https://github.com/google-research/bert
+    ''' 
+    
+    '''we are providing an interface to the 
+          extract_features.py script. For more details (i.e.,
+          choice of Transformer layer) refer to the original paper
+    '''
+    ''' we are using mostly the default parameters for BERT 
+    '''
+    _batch_size = 8
+    _layers = '-1,-2,-3,-4'
+    _max_seq_length = 128
+    _bert_config_file = 'bert_config.json'
+    _do_lower_case = True
+    _num_tpu_cores = 8
+    _master = None
+    _use_tpu = False
+    _init_checkpoint = 'bert_model.ckpt'
+    _use_one_hot_embeddings = False
+    _method = 'bert'
+    _model = 'bert_uncased_L-12_H-768_A-12'
```

Comment:
The more that these config variables overlap in meaning between models the better.  Usually we use these to indicate variables that differentiate the behavior between similar extractors, rather than to specify config variables that only apply to a single extractor.